File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * @author HJX
 * @create 2023-10-23 0:30
 */
public interface BrandService {
    public List<TbBrand> findAll();

    public PageResult findPage(int pageNum,int pageSize);

//    增加
    public void add(TbBrand brand);
    //修改
    public void update(TbBrand brand);
    /*根据id获取实体*/
    public TbBrand findOne(Long id);
    /*批量删除*/
    public void delete(Long [] ids);
    /*分页*/
    public PageResult findPage(TbBrand brand,int pageNum,int pageSize);

    /*下拉列表数据*/
    public List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务层
app.service("brandService", function ($http) {//构建前端服务层
    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    }
    this.findPage = function (page, rows) {
        return $http.get('../brand/findPage.do?page=' + page + '&rows=' + rows);
    }
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity)
    }
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity)
    }
    this.search = function (page, rows, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + "&rows=" + rows, searchEntity)
    }
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id)
    }
    this.dele = function (Ids) {
        return $http.get('../brand/delete.do?ids=' + Ids)
    }
    this.selectOptionList=function () {
        return $http.get('../brand/selectOptionList.do')
    }
})